hydra-node/src/Hydra/Chain/Direct/Wallet.hs

```diff
@@ -398,11 +400,14 @@ coverFee_ pparams systemStart epochInfo lookupUTxO walletUTxO partialTx@Validate
     sortedInputs = sort $ toList initialInputs
     sortedFinalInputs = sort $ toList finalInputs
     differences = List.findIndices (not . uncurry (==)) $ zip sortedInputs sortedFinalInputs
-    adjustOne (ptr@(RdmrPtr t idx), (d, _exUnits))
-      | fromIntegral idx `elem` differences =
-        (RdmrPtr t (idx + 1), (d, executionUnitsFor ptr))
-      | otherwise =
-        (ptr, (d, executionUnitsFor ptr))
+
+    adjustOne (ptr, (d, _exUnits)) =
```

Comment:
nitpick: this is a change of formatting unrelated to the goal of the PR, although I agree not having the `ptr@` pattern makes it more legible.

Reply:
No. This was a problem I fixed on the way. The wallet was shuffling redeemers where it shouldn't. Namely it was doing `idx + 1` on a `Mint` redeemer.. which confused me and I also think @KtorZ when seeing the error (bigger comment thread above). The commit introducing this also explains it.

Reply:
Ah! Good one... we were adjusting redeemers irrespective of their target :s This redeemer pointers are a bliss to work with :relaxed: